### PR TITLE
[core] Rate Limit Guild/Vendor Shop through PacketGuard

### PR DESCRIPTION
**Pull request by 0x05010705**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Adds `0x0AA` and `0x083` to Packet Guard, rate-limited to 1s
  - This prevents players from being able to packet-inject vendors and guild shops to rapid purchase items/limited-supply items.

Example Test at 5s: https://youtu.be/6ae4O8YfHV8

## Steps to test these changes

1 - Login with any character
2 - Go to a guild shop, purchase an item from the Guild Shop
3 - Attempt to purchase again with 1s
Result: The purchase within the time limit is blocked and a message is generated in the map server console.

4 - Wait till the 1s timer has elapsed and make another purchase from the guild shop.
Result: The purchase will now go through since the minimum wait time has been observed.

### CODE REVIEW

**Comment by 0x05010705** [comment]
I would add, this could be expanded to *all* shops and converted to be more generalized to prevent both Guild and Vendor shop injections.  If that is desired, let me know and I'll update things to be more general and apply the checks into 0x083 too.

**Review by Xaver-DaRed — APPROVED** [review]
If you can extend this to other shop types, I would say do so.

**Comment by WinterSolstice8** [comment]
sounds good to me, and you can make it a variable on CCharEntity* as to not waste execution time on localvars

**Comment by WinterSolstice8** [comment]
Also, default value of 250ms is probably close to what the client enforces. I know there is a scheduler for things like the outgoing equip packets

**Comment by 0x05010705** [comment]
Going to simplify this to just PacketGuard at this point.

This keeps all the rate-limiting centralized to a single spot so that if the implementation is changed at some point, there is a central list of packets that need the rate-limit imposed vs hunting through individual packets for edge case rate-limits.

The general consensus for most p servers is that packet injected purchases are bad (even if SE doesnt care), so we can just set a low timer (1s).  Will test this and push those changes / update the PR text.

File: src/map/packet_guard.cpp [hunk 1]
```diff
@@ -85,6 +85,8 @@ namespace PacketGuard
         ratelimitList[0x017] = 1s; // Invalid NPC Information Response
         ratelimitList[0x03B] = 1s; // Mannequin Equip
         ratelimitList[0x05D] = 2s; // Emotes
+        ratelimitList[0x083] = 1s; // Vendor Shop Purchase
```
Comment (WinterSolstice8) [inline src/map/packet_guard.cpp]: Won't these effect normal players too? 1s re-buy time is probably too high for this, i'd set it to 250ms

**Review by WinterSolstice8 — APPROVED** [review]

File: src/map/packet_guard.cpp (hunk 1 above)
Comment (0x05010705) [reply src/map/packet_guard.cpp]: Yea, this is why I was thinking the setting was better in some ways, but this is still the cleaner implementation.
The "average rate of buying" for any given player is arbitrary at best.

Updated to 250ms for now.

**Review by WinterSolstice8 — APPROVED** [review]
